functorch/_src/aot_autograd.py

```diff
@@ -567,7 +1105,106 @@ def backward(ctx, *flat_args):
-        return CompiledFunction.apply(*remove_dupe_args(args))
+        # Step 1: remove dupe args
+        no_dupe_args = remove_dupe_args(args)
+
+        # Step 2: remove aliased inputs that are mutated, replace with synthetic bases
+        # Only happens if our graph mutates an input that aliases another input.
+        if CompiledFunction.synthetic_base_info is not None:
+            # Given: the original args, including at least one pair of inputs that are aliased
+            # and get subsequently mutated.
+            # Generate: the updated args, including (potentially multiple) synthetic bases
+            # that replace the views. The input views are regenerated manually in the compiled function.
+            # TODO: think harder about what happens if (a view of) one of these mutated input views is ALSO returned
+            new_inputs, metadata = merge_view_inputs(no_dupe_args, CompiledFunction.fw_metadata.mutated_input_info)
+            # We're just re-running the original-args-to-synthetic-base transformation
+            # that we ran during compilation.
+            # This returns metadata that we use during tracing to recover the input views,
+            # which we don't actually need at runtime.
+            assert metadata is not None
+            no_dupe_args_with_synthetic_bases = new_inputs
+        else:
+            no_dupe_args_with_synthetic_bases = no_dupe_args
+
+        outs = CompiledFunction.apply(*no_dupe_args_with_synthetic_bases)
+
+        # Step 3: After running the compiled fw, apply updates to mutated inputs
+        if CompiledFunction.num_mutated_inputs > 0:
+            assert len(outs) == CompiledFunction.num_mutated_data_inputs + CompiledFunction.num_non_aliased_outs
+            # Calling convention: mutated inputs show up first.
+            updated_inputs = outs[:CompiledFunction.num_mutated_data_inputs]
+            fw_outs = outs[CompiledFunction.num_mutated_data_inputs:]
+            curr_mutated_inpt_idx = 0
+            curr_metadata_only_mutated_inpt_idx = 0
+            for inpt_idx, mutation_type in enumerate(CompiledFunction.fw_metadata.mutated_input_info):
+                if mutation_type == MutationType.none:
+                    continue
+                original_inpt = no_dupe_args[inpt_idx]
+                if mutation_type == MutationType.metadata_only:
+                    expected_metadata = CompiledFunction.fw_metadata\
+                        .metadata_mutation_input_info[curr_metadata_only_mutated_inpt_idx]
+                    curr_metadata_only_mutated_inpt_idx += 1
+                    original_inpt.as_strided_(
+                        expected_metadata.size(), expected_metadata.stride(), expected_metadata.storage_offset())
+                else:
+                    updated_inpt = updated_inputs[curr_mutated_inpt_idx]
+                    curr_mutated_inpt_idx += 1
+                    # TODO: handle resize_() on inputs to a larger size.
```

Comment:
This will be silent correctness if I turn off the analysis pass today, so I'll either fix resize_() or graph break in dynamo before landing.